### PR TITLE
Add REST endpoint for getting node settings

### PR DESCRIPTION
**Pull request by CodeSandwich**
Fixes https://github.com/input-output-hk/jormungandr/issues/410 and probably https://github.com/input-output-hk/jormungandr/issues/86

### CODE REVIEW

File: jormungandr/src/leadership/task.rs [hunk 1]
```diff
@@ -67,17 +70,21 @@ impl Task {
             fragment_pool,
             epoch_receiver,
             block_message,
+            stats_counter,
         }
     }
 
     pub fn start(self) -> impl Future<Item = (), Error = ()> {
-        let handle_logger = self.logger.clone();
+        let params = HandleLeadershipParams {
```
Comment (mzabaluev) [inline jormungandr/src/leadership/task.rs]: Can `Task` and `HandleLeadershipParams` simply reuse a common structure?

File: jormungandr/src/stats_counter.rs [hunk 2]
```diff
@@ -39,11 +41,27 @@ impl StatsCounter {
...
-    pub fn get_uptime_sec(&self) -> u64 {
+    pub fn uptime_sec(&self) -> u64 {
         self.stats.start_time.elapsed().as_secs()
     }
+
+    pub fn set_slot_start_time(&self, time: SystemTime) {
+        self.stats
+            .slot_start_time
+            .write()
+            .unwrap_or_else(PoisonError::into_inner)
```
Comment (mzabaluev) [inline jormungandr/src/stats_counter.rs]: Does this really need to survive a poisoned mutex?

Comment (CodeSandwich) [reply jormungandr/src/stats_counter.rs]: The mutex should never be poisoned, but even if it is, the underlying data is not corruptable with panics. Could be this or `.unwrap`, doesn't matter. I personally like avoiding statements capable of panicking.

File: jormungandr/src/stats_counter.rs [hunk 3]
```diff
@@ -12,6 +12,7 @@ struct StatsCounterImpl {
     tx_recv_cnt: AtomicUsize,
     block_recv_cnt: AtomicUsize,
     start_time: Instant,
+    slot_start_time: RwLock<Option<SystemTime>>,
```
Comment (vincenthz) [inline jormungandr/src/stats_counter.rs]: RwLock for this seems totally overkill

Comment (CodeSandwich) [reply jormungandr/src/stats_counter.rs]: I couldn't cram SystemTime into a single Atomic (it takes 128 bits) and the data in this structure must be both Sync and internally-mutable :(

Comment (vincenthz) [reply jormungandr/src/stats_counter.rs]: I would put it in 2 atomic u64. it really doesn't matter is someone grab this value in the middle of an update, and not totally sure we need to keep the nanoseconds precision anyway

Comment (NicolasDP) [reply jormungandr/src/stats_counter.rs]: In jormungandr there is `SecondsSinceUnixEpoch` which is a `u64` already. Does not contain the nanoseconds precision. And it provides compatibility back into/from `SystemTime`.

File: doc/jcli/rest.md [hunk 4]
```diff
@@ -206,3 +206,34 @@ value: 990
...
   it is possible this value is not set if there is no delegation certificate
   sent associated to this account.
+
+
+## Node settings
+
+Fetches node settings
+
+```
+jcli rest v0 node settings get <options>
```
Comment (sjmackenzie) [inline doc/jcli/rest.md]: shouldn't have `node` in the command, also `jcli rest v0 settings get --output-format yaml -h <host>` doesn't work.

Comment (CodeSandwich) [reply doc/jcli/rest.md]: Good one about the docs.

But I can't reproduce JCLI not working:
```
./jcli rest v0 settings get --output-format yaml -h http://127.0.0.1:8443/api
---
block0Hash: 8d94ecfcc9a566f492e6335858db645691f628b012bed4ac2b1338b5690355a7
block0Time: "2019-07-09T12:32:51+00:00"
consensusVersion: bft
currSlotStartTime: ~
fees:
  certificate: 4
  coefficient: 1
  constant: 2
maxTxsPerBlock: 100
```
What's the failure on your computer?

Comment (sjmackenzie) [reply doc/jcli/rest.md]: Yeah I edited it out because I forgot to append the `/api` :-) sorry about the noise.

Comment (CodeSandwich) [reply doc/jcli/rest.md]: Whew, good to hear :)

**Pull request merged**